.github/workflows/ci-cd.yaml

```diff
@@ -0,0 +1,72 @@
+---
+name: CI Build
+on:
+  schedule:
+    - cron: '28 0 * * 1'
+  push:
+    branches: ['master']
+  pull_request:
+    branches: ['master']
+
+jobs:
+  terraform:
+    name: "Terraform"
+    runs-on: ubuntu-latest
+    steps:
+      - name: "Checkout ${{ github.ref }} ( ${{ github.sha }} )"
+        uses: actions/checkout@v2
+
+      - name: "Setup Google SDK"
+        uses: GoogleCloudPlatform/github-actions/setup-gcloud@master
+        if: ${{ github.event_name == 'push' }}
+        with:
+          version: '307.0.0'
+          project_id: ${{ secrets.GCP_PROJECT_ID }}
+          service_account_key: ${{ secrets.GCP_SA_KEY }}
+          export_default_credentials: true
+
+      - run: gcloud info
+        if: ${{ github.event_name == 'push' }}
+
+      - name: "Setup Terraaform"
+        uses: hashicorp/setup-terraform@v1
+        with:
+          terraform_version: 0.12.25
+
+      - name: Terraform Init
+        run: terraform init
+        working-directory: terraform
+
+      - name: Terraform Validate
+        run: terraform validate -no-color
+        working-directory: terraform
+
+      - name: Terraform Plan
+        if: github.event_name == 'push'
+        run: terraform plan -no-color
```

Comment:
```suggestion
        run: terraform plan -no-color -input=false
```

Reply:
What do you think about saving `plan` result to file and the using it in `apply` step?
```
terraform plan -out=tfplan -input=false
terraform apply -input=false tfplan
```

Reply:
I prefer not to complicate the scripts.I do not see much benefit in this case.